threads.js

```diff
@@ -164,7 +164,63 @@ ChildThread.prototype.destroyProcess = function() {
   }
 };
 
-},{"./emitter":4,"./utils":8}],3:[function(require,module,exports){
+},{"./emitter":5,"./utils":10}],3:[function(require,module,exports){
+
+'use strict';
+
+var Emitter = require('./emitter');
+var utils = require('./utils');
+
+module.exports = ClientStream;
+
+function ClientStream(client, id) {
+  this._client = client;
+  this._id = id;
+  this._closed = utils.deferred();
+  this._emitter = new Emitter();
+}
+
+ClientStream.prototype = {
+  get closed() {
+    return this._closed.promise;
+  },
+
+  onabort_: function(data) {
+    this._closed.reject(data);
+  },
+
+  onclose_: function(data) {
+    this._closed.resolve(data);
+  },
+
+  ondata_: function(data) {
+    this._emitter.emit('data', data);
+  },
+
+  listen: function(callback) {
+    var emitter = this._emitter;
+    emitter.on('data', callback);
+    return {
+      cancel: function() {
+        emitter.off('data', callback);
+      }
```

Comment:
uhm.. now that I'm thinking about it, it would be *better* to have an `unlisten` or `removeListener` method instead of returning a new object.